internal/libs/clist/clist.go

```diff
@@ -72,27 +72,9 @@ func (e *CElement) NextWait() *CElement {
 	}
 }
 
-// Blocking implementation of Prev().
-// May return nil iff CElement was head and got removed.
-func (e *CElement) PrevWait() *CElement {
-	for {
-		e.mtx.RLock()
-		prev := e.prev
-		removed := e.removed
-		signal := e.prevWaitCh
-		e.mtx.RUnlock()
-
-		if prev != nil || removed {
-			return prev
-		}
-
-		<-signal
-	}
-}
-
-// PrevWaitChan can be used to wait until Prev becomes not nil. Once it does,
+// prevWaitChan can be used to wait until Prev becomes not nil. Once it does,
 // channel will be closed.
-func (e *CElement) PrevWaitChan() <-chan struct{} {
+func (e *CElement) prevWaitChan() <-chan struct{} {
```

Comment:
I think that this method can also be deleted since it just exposes access the `prevWaitCh` member field to the caller. Anything accessing this method can just access the field directly, which this code does in `prevWait`.

I see that it's used in the test, but since it's fully internal, I think it can just be removed from the test as well.